vllm/worker/neuron_model_runner.py

```diff
@@ -225,13 +254,48 @@ def prepare_model_input(
             seq_lens,
             self.device,
             self.pin_memory,
-            generators=self.get_generators(finished_requests_ids))
+            generators=self.get_generators(finished_requests_ids),
+        )
 
-        return ModelInputForNeuron(input_tokens=input_tokens,
-                                   input_positions=input_positions,
-                                   input_block_ids=input_block_ids,
-                                   sampling_metadata=sampling_metadata,
-                                   multi_modal_kwargs=multi_modal_kwargs)
+        return ModelInputForNeuron(
+            input_tokens=input_tokens,
+            input_positions=input_positions,
+            input_block_ids=input_block_ids,
+            sampling_metadata=sampling_metadata,
+            multi_modal_kwargs=multi_modal_kwargs,
+        )
+
+    def _update_neuron_generation_config(self, sampling_metadata):
+        # Update Neuron Genetation Config
+        current_generation_config = self.model_config.generation_config
+        assert current_generation_config is not None, (
+            f"Failed to update generation_config, "
+            f"current generation config is {current_generation_config}")
+
+        top_k = current_generation_config.top_k.copy()
+        top_p = current_generation_config.top_p.copy()
+        temperature = current_generation_config.temperature.copy()
+        for index, sequence_group_to_sample in enumerate(
+                sampling_metadata.seq_groups):
+            top_k[index] = sequence_group_to_sample.sampling_params.top_k
+            top_p[index] = sequence_group_to_sample.sampling_params.top_p
+            temperature[index] = (
+                sequence_group_to_sample.sampling_params.temperature)
+
+        # Only call update the generation config is the new config is different.
+        # This avoids calling update_generation_config for every token within
+        # the same sequence.
+        if (top_k != current_generation_config.top_k
+                or top_p != current_generation_config.top_p
+                or temperature != current_generation_config.temperature):
+            current_generation_config.top_k = top_k
+            current_generation_config.top_p = top_p
+            current_generation_config.temperature = temperature
+            self.model_config.generation_config = copy.deepcopy(
+                current_generation_config)
```

Comment:
as this is part of `execute_model` function, `deepcopy` could introduce unnecessary overhead.

Reply:
ack, remove the deepcopy as the config duplication is already done during the `self.model.model.update_generation_config(current_generation_config)`